src/local/room_xy.rs

```diff
@@ -67,6 +71,19 @@ pub struct RoomXY {
 }
 
 impl RoomXY {
+    /// Create a new `RoomXY` from a pair of `RoomCoordinate`.
+    #[inline]
+    pub fn new(x: RoomCoordinate, y: RoomCoordinate) -> Self {
```

Comment:
I suppose it's convention to have it, but it also feels weird given you can always do `RoomXY { x, y }` because of the pub fields.

Reply:
Honestly, future-proofing + convention following + "it feels really weird to build the struct in-place instead of using ::new"